tests/unit/apps_planning/test_apps_plan.py

```diff
@@ -0,0 +1,34 @@
+#  Copyright 2023 Canonical Limited
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Test all sample plans."""
+from unittest.mock import patch
+
+import pytest
+
+from cou.commands import CLIargs
+from cou.steps.analyze import Analysis
+from cou.steps.plan import generate_plan
+
+
+@pytest.mark.asyncio
+@patch("cou.utils.nova_compute.get_instance_count", return_value=0)
```

Comment:
I suspect keeping the input yamls as close to the production cloud exports would be quite adequate. Whether some hypervisors are or are not running instances is a "runtime" conditions and is not part of the model. Keeping this mocked return here allows us to more easily test what happens if none/some/all hypervisors are empty.

Reply:
Currently, we are not storing this information directly with `Application` object, but it can be found in logs I think. However, I do see how it would be useful to define it in the yaml file, since it will be run multiple times (once in stage of planning and once during actual upgrade). 
It can be defined in yaml file just for planning stage, what do you say?